src/XMLexport.cpp

```diff
@@ -378,8 +386,18 @@ bool XMLexport::writeHost(Host* pHost)
             pHost->modulesToWrite.clear();
             while (it.hasNext()) {
                 it.next();
-                writeTextElement("key", it.key());
+
                 QStringList entry = it.value();
+                writeStartElement("key");
+                // Added in 1.002 to make the representation easier/quicker to
```

Comment:
I don't think this is a worthwhile change to cause pain to people. This is what people think of these trivial changes in practice:

![selection_267](https://user-images.githubusercontent.com/110988/33873294-b67b29bc-df1a-11e7-8354-63cf5869c830.png)

Reply:
The structure chosen to represent the XML data is less than perfect and there are places where the way the data is stored is fragile: splitting repeating elements, for example:

            <mInstalledModules>
                <key>WoTMUD</key>
                <filepath>/home/stephen/mudlet-data/packages/WoTMUD_0_35.mpackage</filepath>
                <globalSave>1</globalSave>
                <priority>-5</priority>
                <key>moduleAndPackageMonitor</key>
                <filepath>/home/stephen/mudlet-data/packages/moduleAndPackageMonitor.zip</filepath>
                <globalSave>0</globalSave>
                <priority>0</priority>
            </mInstalledModules>

which I'd like to see reduced to:

            <installedModules>
                <module name="WoTMUD" isSynced=1 priority=-5>/home/stephen/mudlet-data/packages/WoTMUD_0_35.mpackage</module>
                <module name="moduleAndPackageMonitor">/home/stephen/mudlet-data/packages/moduleAndPackageMonitor.zip</module>
            </installedModules>

and say:

            <Trigger isActive="yes" isFolder="no" isTempTrigger="no" isMultiline="no" isPerlSlashGOption="no" isColorizerTrigger="yes" isFilterTrigger="no" isSoundTrigger="no" isColorTrigger="no" isColorTriggerFg="no" isColorTriggerBg="no">
                <name>movement failure</name>
                <script>	List:popleft(wotmudmapper.queue)</script>
                <triggerType>0</triggerType>
                <conditonLineDelta>0</conditonLineDelta>
                <mStayOpen>0</mStayOpen>
                <mCommand></mCommand>
                <packageName></packageName>
                <mFgColor>#ffaa00</mFgColor>
                <mBgColor>#000000</mBgColor>
                <mSoundFile></mSoundFile>
                <colorTriggerFgColor>#000000</colorTriggerFgColor>
                <colorTriggerBgColor>#000000</colorTriggerBgColor>
                <regexCodeList>
                    <string>Alas, you cannot go that way...$</string>
                    <string>You can't ride in there.$</string>
                    <string>An (.*) blocks the way.$</string>
                    <string>A (.*) blocks the way.$</string>
                    <string>No way!  You're fighting for your life!$</string>
                    <string>You need a boat to go there.$</string>
                    <string>Your mount is too exhausted.$</string>
                    <string>You can't ride there on a horse!$</string>
                    <string>Are you crazy? You can't just walk out of a second story window!$</string>
                    <string>It is pitch black...$</string>
                    <string>Nah... You feel too relaxed to do that..$</string>
                    <string>You would need to swim there, you can't just walk it.$</string>
                    <string>PANIC!  You couldn't escape!$</string>
                    <string>You can't see a damned thing, you're blinded!$</string>
                    <string>You'll have to take your chances and JUMP OUT!</string>
                    <string>Your mount is engaged in combat!$</string>
                    <string>You shudder at the concept of crossing water.$</string>
                    <string>The iron golem blocks the path of (.+)$</string>
                    <string>You are too exhausted.$</string>
                    <string>Berserk! Death! Death! Fight to the death!$</string>
                    <string>Maybe you should get on your feet first\?$</string>
                    <string>You try, but the mind-numbing cold prevents you..$</string>
                    <string>The ground gives way, and you are sent hurtling down the embankment!$</string>
                    <string>You can't ride on water.$</string>
                    <string>In your dreams, or what\?$</string>
                </regexCodeList>
                <regexCodePropertyList>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                    <integer>1</integer>
                </regexCodePropertyList>
            </Trigger>

whereas by making as many things as possible as _attributes_ (with sensible default values to use if they are not found) rather than _sub-elements_ it can be made more compact and easier for both machine and humans to parse:

            <Trigger isActive="yes" type="0" isColorizerTrigger="yes" fgColor="#ffaa00" mBgColor="#000000">
                <name>movement failure</name>
                <script>	List:popleft(wotmudmapper.queue)</script>
                <command />
                <packageName />
                <soundFile />
                <triggerItems>
                    <triggerItem type="regExp">Alas, you cannot go that way...$</triggerItem>
                    <triggerItem type="regExp">You can't ride in there.$</triggerItem>
                    <triggerItem type="regExp">An (.*) blocks the way.$</triggerItem>
                    <triggerItem type="regExp">A (.*) blocks the way.$</triggerItem>
                    <triggerItem type="regExp">No way!  You're fighting for your life!$</triggerItem>
                    <triggerItem type="regExp">You need a boat to go there.$</triggerItem>
                    <triggerItem type="regExp">Your mount is too exhausted.$</triggerItem>
                    <triggerItem type="regExp">You can't ride there on a horse!$</triggerItem>
                    <triggerItem type="regExp">Are you crazy? You can't just walk out of a second story window!$</triggerItem>
                    <triggerItem type="regExp">It is pitch black...$</triggerItem>
                    <triggerItem type="regExp">Nah... You feel too relaxed to do that..$</triggerItem>
                    <triggerItem type="regExp">You would need to swim there, you can't just walk it.$</triggerItem>
                    <triggerItem type="regExp">PANIC!  You couldn't escape!$</triggerItem>
                    <triggerItem type="regExp">You can't see a damned thing, you're blinded!$</triggerItem>
                    <triggerItem type="regExp">You'll have to take your chances and JUMP OUT!</string>
                    <triggerItem type="regExp">Your mount is engaged in combat!$</triggerItem>
                    <triggerItem type="regExp">You shudder at the concept of crossing water.$</triggerItem>
                    <triggerItem type="regExp">The iron golem blocks the path of (.+)$</triggerItem>
                    <triggerItem type="regExp">You are too exhausted.$</triggerItem>
                    <triggerItem type="regExp">Berserk! Death! Death! Fight to the death!$</triggerItem>
                    <triggerItem type="regExp">Maybe you should get on your feet first\?$</triggerItem>
                    <triggerItem type="regExp">You try, but the mind-numbing cold prevents you..$</triggerItem>
                    <triggerItem type="regExp">The ground gives way, and you are sent hurtling down the embankment!$</triggerItem>
                    <triggerItem type="regExp">You can't ride on water.$</triggerItem>
                    <triggerItem type="regExp">In your dreams, or what\?$</triggerItem>
                </triggerItems>
            </Trigger>

and grossly inefficient storage of variables, by definition I think a `VariableGroup` is a table (`valueType == 5`) but I feel sure there is a way to make things more compact I mean it takes 30 lines of text to write just a four number value table:
 
        <VariableGroup>
            <name>wotmud</name>
            <keyType>4</keyType>
            <value></value>
            <valueType>5</valueType>
            <Variable>
                <name>seconds</name>
                <keyType>4</keyType>
                <value>0</value>
                <valueType>3</valueType>
            </Variable>
            <Variable>
                <name>hours</name>
                <keyType>4</keyType>
                <value>9</value>
                <valueType>3</valueType>
            </Variable>
            <Variable>
                <name>days</name>
                <keyType>4</keyType>
                <value>0</value>
                <valueType>3</valueType>
            </Variable>
            <Variable>
                <name>minutes</name>
                <keyType>4</keyType>
                <value>25</value>
                <valueType>3</valueType>
            </Variable>
        </VariableGroup>

and what is the context for that third party moan? I cannot find them on github except as someone with the same gravatar icon and a single user name as their lowercase first name who is totally inactive on here apart from a single registration activity so I guess that list of things is elsewhere...

Reply:
Yes, I agree! The XML structure is inefficient - but what it should be is a complete discussion on its own. It's a question of whenever XML is the best storage format to use at all. Let's tackle this outside this PR, ok? For now making little tweaks is not worthwhile for the pain.

Reply:
I've created https://github.com/Mudlet/Mudlet/issues/1464 to capture your thoughts and address this in a separate scope. I'd like to have this PR leave the storage be so we can have a proper look at it and not piecemeal.